src/pages/Home/Banner/anime.ts

```diff
@@ -2,96 +2,115 @@ import { Mesh } from 'three'
 import { gsap } from 'gsap'
 import { singleton } from '../../../utils/util'
 import { assert } from '../../../utils/error'
+import { CubeOffset, InstancedCubeUnitControl } from './renderUtils'
 
 export const playNewBlockAnime = singleton(_playNewBlockAnime)
 
 const DURATION_TO_PROTRUDE = 5 // duration to protrude
 const DELAY_OF_RIPPLE = 10 // change the distance between edges of a ripple
 
 // eslint-disable-next-line no-underscore-dangle
-function _playNewBlockAnime(cubes: Mesh[], textCubes: Mesh[], dataCubes: Mesh[], createFloatCube: () => Mesh) {
-  return Promise.all(cubes.map(startCubeAnime))
+function _playNewBlockAnime(
+  cubeOffsets: CubeOffset[],
+  getCubeControl: (
+    x: number,
+    z: number,
+  ) =>
+    | (InstancedCubeUnitControl & {
+        isTextCube: boolean
+        isDataCube: boolean
+        isFloatCubeCreator: boolean
+      })
+    | null,
+  createFloatCube: () => Mesh,
+) {
+  return Promise.all(cubeOffsets.map(startCubeAnime))
+
+  function startCubeAnime(offset: CubeOffset): gsap.core.Animation | null {
+    const control = getCubeControl(offset.x, offset.z)
+    if (control == null) return null
+    const { position, onUpdate, isTextCube, isDataCube, isFloatCubeCreator } = control
 
-  async function startCubeAnime(cube: Mesh): Promise<gsap.core.Animation> {
     const textCubeFirstUpDistance = 40
-    const ySpeedPerSecond = 150
-    const getMoveTime = (distance: number) => distance / ySpeedPerSecond
+    const ySpeedPerSecond = 75
+    const getMoveTime = (distance: number, multiply = 1) => distance / ySpeedPerSecond / multiply
 
-    if (textCubes.includes(cube)) {
+    if (isTextCube) {
       return gsap
         .timeline()
-        .to(cube.position, {
-          y: cube.position.y + textCubeFirstUpDistance * 2,
+        .to(position, {
+          y: position.y + textCubeFirstUpDistance * 2,
           duration: getMoveTime(DURATION_TO_PROTRUDE * 5),
+          onUpdate,
         })
```

Comment:
The suggestion below may not match the design, if not, please ignore. 
For the wave effect in the design, setting the `ease` of the text cubes' timeline to `sine.out sine.inOut sine.in`, and increase `duration` of the normal cubes' timeline may make the movement more similar to waves

Reply:
The design expects a rhythm similar to a heartbeat, and I didn't quite understand exactly what kind of effect it would be better to arrive at.

I tried to modify it according to what you said, but maybe the parameters are not right, so the effect is not very good.

You can provide a recorded screen to the comments, and then give the designer to see the effect, and then propose a PR based on this branch if it is confirmed that the effect is better.